src/Files.App/Utils/Taskbar/SystemTrayIcon.cs

```diff
@@ -266,18 +266,17 @@ private void ShowContextMenu()
 		private void OnLeftClicked()
 		{
 			// Prevents duplicate launch
-			if (_timer?.IsRunning ?? false)
+			if (DateTime.Now - _lastLaunchDate < TimeSpan.FromSeconds(1))
 				return;
 
 			if (Program.Pool is not null)
 			{
-				_timer ??= DispatcherQueue.GetForCurrentThread().CreateTimer();
-				_timer.Interval = TimeSpan.FromSeconds(1);
-				_timer.IsRepeating = false;
-				_timer.Start();
+				_lastLaunchDate = DateTime.Now;
 
 				_ = Launcher.LaunchUriAsync(new Uri("files-uwp:"));
 			}
+			else
+				MainWindow.Instance.Activate();
```

Comment:
This will bring the already open window to the foreground.

Reply:
Thanks!